packages/salesforce/src/queryRecordFactory.ts

```diff
@@ -164,6 +164,22 @@ export class RecordFactory {
             ?? name;
     }
 
+    private static transformValue(value: string | number | boolean | undefined): any {
```

Comment:
The transformer will only be sued for the proxy which makes the choice to use a proxy or affect the object returned in a significant way. I don't think that is desirable.

Is there a way to also do this for the none proxied record?